### PR TITLE
welcome email sending

### PR DESCRIPTION
**Pull request by nickysemenza**

### CODE REVIEW

File: app/Http/Controllers/ProfileController.php [hunk 1]
```diff
@@ -21,6 +25,47 @@ public function me()
...
+        }
+
+        //todo: integrity constraint check for email and nickname
+        $user->save();
+
+
+        //check if email changed
+        if($oldEmail != $user->email) {
+
+            if($oldEmail == null) {
```
Comment (harrischristiansen) [inline app/Http/Controllers/ProfileController.php]: Add space `if ($oldEmail == null)`

File: app/Http/Controllers/ProfileController.php [hunk 2]
```diff
@@ -21,6 +25,47 @@ public function me()
...
+                $user->$key = $value;
+            }
+        }
+
+        //todo: integrity constraint check for email and nickname
+        $user->save();
+
+
+        //check if email changed
+        if($oldEmail != $user->email) {
```
Comment (harrischristiansen) [inline app/Http/Controllers/ProfileController.php]: Add space `if ($oldEmail != $user->email)`

File: app/Http/Controllers/ProfileController.php [hunk 3]
```diff
@@ -21,6 +25,47 @@ public function me()
...
+
+            if($oldEmail == null) {
+                //setting email for first time
+                Mail::to($user)->send(new WelcomeEmail($user));
+                //todo: confirmation?
+            } else if($user->email = "") {
```
Comment (harrischristiansen) [inline app/Http/Controllers/ProfileController.php]: Add space `} else if ($user->email = "") {`

File: database/factories/UserFactory.php [hunk 4]
```diff
@@ -14,13 +14,12 @@
...
-    static $password;
 
     return [
-        'name'           => $faker->name,
-        'email'          => $faker->unique()->safeEmail,
-        'password'       => $password ?: $password = bcrypt('secret'),
-        'remember_token' => str_random(10),
+
+        'nickname' => $faker->firstName,
+//        'email' => $faker->unique()->safeEmail,
```
Comment (harrischristiansen) [inline database/factories/UserFactory.php]: If you want, for now, we can have all fake accounts share the email `ccfake@purduecs.com`.

Once we buy a domain, we can use that domain.

Comment (nickysemenza) [reply database/factories/UserFactory.php]: this is for unit tests and such, so the emails don't have to be real

**Pull request merged**